pageserver/src/tenant/timeline.rs

```diff
@@ -3607,7 +3607,7 @@ impl Timeline {
                         if let Ok((b_key, b_lsn, _)) = b {
                             match a_key.cmp(b_key) {
                                 Ordering::Less => true,
-                                Ordering::Equal => a_lsn <= b_lsn,
+                                Ordering::Equal => a_lsn < b_lsn,
```

Comment:
FTR this entire `match` can be reduced to use comparison on tuples, to use `(a_key, a_lsn) < (b_key, b_lsn)`. Same goes for the other match down below.